ESQL: Docs for `documents_found` (#147989)
elastic/elasticsearch: docs/reference/query-languages/esql/esql-rest.md

You will also receive the async ID and running status in the `X-Elasticsearch-Async-Id` and `X-Elasticsearch-Async-Is-Running` HTTP headers of the response, respectively.
Useful if you use a tabular text format like `txt`, `csv` or `tsv`, as you won't receive those fields in the body there.
::::

## `documents_found` and `values_loaded` [esql-rest-documents-found]

> 

Reviewer: ``````suggestion
```{applies_to}
stack: ga 9.5
```
``````

Reviewer: This is the only really important edit :)

Author: :+1:

This was added quite a bit before 9.5, I just didn't document it. I'll update the ga release to whatever it was.

In addition to {{es}}'s traditional `took` time, {{esql}} returns `documents_found` and `values_loaded` which you can think
of as rough proxies for how much effort Elasticsearch had to expend to run the query. Generally, bigger numbers mean the
query took more effort.

`documents_found` are the number of documents that we had to find to run the query. If this is low, Elasticsearch was able
to effectively use its search index to return results. A few queries, like `FROM idx | STATS COUNT(*)` can run without

> looking at documents at all. They cheekily report one "found" document per count.

Reviewer: ```suggestion
looking at documents at all. They only report one "found" document per count.
```

I love the whimsy personally, but it's jarring when it's not a general tone across all the docs

But feel free to ignore, it's your call 😄

> `values_loaded` are the number of values that we had to load to run the query. If Elasticsearch must load one value per

Reviewer: ```suggestion
`values_loaded` are the number of values loaded to run the query. If Elasticsearch must load one value per
```

> document, like it would for `FROM idx | STATS BY DATE_TRUNC(1 hour, @timestamp)`, this'll be the same as `documents_found`.

Reviewer: ```suggestion
document, like it would for `FROM idx | STATS BY DATE_TRUNC(1 hour, @timestamp)`, this will be the same as `documents_found`.
```

> If Elasticsearch had to load many fields for each document, it'll be many times `documents_found`. Again, bigger numbers

Reviewer: ```suggestion
If Elasticsearch had to load many fields for each document, it will be many times `documents_found`. Again, bigger numbers
```

*generally* mean the query took more effort.

Plenty of queries will still be fast even though `documents_found` and `values_loaded` are high. Some queries will be slow

> even with a low `documents_found` and `values_loaded`. Because:

Reviewer: ```suggestion
even with a low `documents_found` and `values_loaded` for various reasons:
```

> * The query was super fast. Or slow!

Reviewer: ```suggestion
* The query was super fast or slow.
```

* Some values load very fast, like `@timestamp`, and other values load much more slowly, like `text` fields.
* Some functions are quite fast, like `+`, `DATE_TRUNC`, `BUCKET`, etc.

> * Some commands are quite heavy, like `GROK`.

Reviewer: not sure if it makes sense to be more quantitative than "fast", "slow" and "heavy" here? 🤷

Author: This whole section is a bit fluffy. Heavy is probably fine.

You will also receive the async ID and running status in the `X-Elasticsearch-Async-Id` and `X-Elasticsearch-Async-Is-Running` HTTP headers of the response, respectively.
Useful if you use a tabular text format like `txt`, `csv` or `tsv`, as you won't receive those fields in the body there.
::::

> ## `documents_found` and `values_loaded` [esql-rest-documents-found]

Reviewer: ```suggestion
## Data volume indicators
```

@nik9000 Overlooked the heading initially, which is a bit literal. Consider this a spitball.

Other ideas:

```
## Proxies for query and fetch effort
## Query intensity indicators
## Execution indicators
```

none quite hit the mark, because it's hard to find a nice, clean abstraction for "effort" that covers both loading and querying but you get the drift :)

maybe even _**cost**_ indicators?  because, say, on serverless search and ingest are billed by consumption, so higher documents_found and values_loaded would directly translate to higher costs.
